### PR TITLE
Pin upstream 2.0 release

### PR DESCRIPTION
**Pull request by steventk-g**
* Pin to upstream release branch
* Modify circleCI logic to use release branch for build
* Add version selector logic

### CODE REVIEW

**Comment by JackCaoG** [comment]
There seems to be duplicate commit in this pr

**Review by JackCaoG — APPROVED** [review]

**Comment by steventk-g** [comment]
Resolved the duplication issue. Thanks Jack!

**Review by vanbasten23 — APPROVED** [review]

**Comment by JackCaoG** [comment]
I am not sure why `cat: xla/third_party/tensorflow/third_party/llvm/workspace.bzl: No such file or directory` is failing, I thought it only got trigger if bazel stuff is in the branch.

anyway I think https://github.com/pytorch/xla/pull/4632/commits/5e3a756dcd1d02cf154fb01fb6005bc356393480 should solve that issue.

File: .circleci/download_llvm_raw.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 function download_llvm_raw_archive() {
   # Extract the xla pinned tensorflow/third_party/llvm version,
   # avoid doing this manually.
-  git clone --recursive --quiet https://github.com/pytorch/xla.git
+  git clone --recursive --quiet --branch r2.0 https://github.com/pytorch/xla.git
```
Comment (vanbasten23) [inline .circleci/download_llvm_raw.sh:7]: I wonder if it's necessary and also if we should do the same for upstream pytorch.

Comment (vanbasten23) [reply .circleci/download_llvm_raw.sh:7]: nvm, i saw our chat.

**Pull request merged**